### PR TITLE
Local provider: store metadata file with custom metadata

### LINKED ISSUE #33: Suggestion: store metadata file for local file provider with custom metadata
We're currently in the process of migrating to this library from a home-grown abstraction layer for storage providers, and we're currently relying on being able to store custom metadata for both `cloud` and `local` stores (to be truly independent from cloud support), however, I see that the `UpdateBlobPropertiesAsync` for the local provider simply returns a completed task, effectively making it a no-op.

What I suggest is the following:

Once this method is called, store a `containerName` + `blobName` `.meta` (or similar) where this custom information is stored. It could be in json format or what-not.

When calling the `GetBlobDescriptor` and similar methods that return meta-data about the object, then check if a `.meta` file exists and merge the content that is "automatically guessed" with the custom data stored in the `.meta` file.

This would allow us to not maintain additional code when using the local provider.

And finally, would you accept an MR with the suggested changes?

### PR DESCRIPTION
**Pull request by nover**
This PR will:

- **add** `System.Text.Json` for working with json data
- **update** `UpdateBlobProperties*` to store a `filename-meta.json` file with the content from passed `BlobProperties` 
- **update** `GetBlobDescriptor*` and `List` to merge "guessed" metadata with a stored metadata file
- **update** `GetBlobDescriptor*` to also return the `Metadata:dictionary<string,string>` if stored in the metadata file 
- **update** delete/copy/move operations to work on stored metadata
- **refactor** some internals to reduce code-duplication

Closes #33

### CODE REVIEW

**Comment by nover** [comment]
@ericgreenmix any chance of getting this reviewed and merged? :)

**Comment by ericgreenmix** [comment]
Yes, sorry for the delay. Been very busy this past week. I will try to get to this in the next day

File: src/TwentyTwenty.Storage.Local/LocalStorageProvider.cs [hunk 1]
```diff
@@ -244,21 +264,100 @@ public Task<IList<BlobDescriptor>> ListBlobsAsync(string containerName)
...
+                }
+
+                descriptor.Security = meta.Security;
+                descriptor.Metadata = meta.Metadata;
+            }
+        }
+
+        private BlobDescriptor GetFileMetaIfExists(string fullPathToBlob)
+        {
+            return GetFileMetaIfExistsAsync(fullPathToBlob).ConfigureAwait(false).GetAwaiter().GetResult();
```
Comment (ericgreenmix) [inline src/TwentyTwenty.Storage.Local/LocalStorageProvider.cs:346]: Lets create a non-async version of GetFileMetaIfExists, so we don't have to use `GetResult()`. `StreamReader` has a `ReadToEnd()` method, to replace the async one.

File: src/TwentyTwenty.Storage.Local/LocalStorageProvider.cs [hunk 2]
```diff
@@ -244,21 +264,100 @@ public Task<IList<BlobDescriptor>> ListBlobsAsync(string containerName)
...
             {
                 throw ex.ToStorageException();
             }
         }
-
+        
         public void UpdateBlobProperties(string containerName, string blobName, BlobProperties properties)
         {
-            return;
+            UpdateBlobPropertiesAsync(containerName, blobName, properties).ConfigureAwait(false).GetAwaiter().GetResult();
```
Comment (ericgreenmix) [inline src/TwentyTwenty.Storage.Local/LocalStorageProvider.cs:280]: Lets create a non-async version of `UpdateBlobProperties`, so we don't have to use `GetResult()`. `StreamWriter` has `Write` and `Flush` methods, to replace the async ones.

**Review by ericgreenmix — COMMENTED** [review]
Just a couple things I saw.

I am working on figuring out why the Azure tests are failing, so we can have the tests all passing again.

**Comment by nover** [comment]
@ericgreenmix 

No problems, I know about the busy days my self :)
your suggestions have been fixed and squashed into the original commit.

As for the failing Azure tests, I believe it's because I'm creating a PR from a fork. I'm having similar problems on some of my repositories - Appveyor does not hand out secrets to PR builds from forks, as it would allow malicious code to steal them.
https://help.appveyor.com/discussions/questions/46970-unable-to-access-secure-environment-variable-in-pull-request

**Comment by ericgreenmix** [comment]
@nover 2.17.1 is now out with these changes. Thanks!

**Pull request merged**